### PR TITLE
chore: address pentest findings & add non GSI e2e tests

### PR DESCRIPTION
**Pull request by AaronZyLee**
#### Description of changes

- Addressed pentest findings
  - Index name format check to avoid injection
  - Remove response URL in log
  - Add allow list when converting string type to boolean or number
- Added construct e2e test for multiple GSI updates + other non GSI updates. The tests are passed locally.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] Any CDK or CloudFormation parameter changes are called out explicitly

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by alharris-at** [comment]
Could you run the e2e suite on this change, as well, since you've added a test?

File: packages/amplify-graphql-api-construct-tests/src/__tests__/amplify-table.test.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    try {
+      await cdkDestroy(projRoot, '--all');
+    } catch (_) {
+      /* No-op */
+    }
+
+    deleteProjectDir(projRoot);
+  });
+
+  ['2.80.0'].forEach((cdkVersion) => {
```
Comment (alharris-at) [inline packages/amplify-graphql-api-construct-tests/src/__tests__/amplify-table.test.ts:27]: No need to specify this. You can just write a test, and no need to pass in the default version to `initCDKProject`.

File: packages/amplify-graphql-api-construct-tests/src/__tests__/amplify-table.test.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      const templatePath = path.resolve(path.join(__dirname, 'backends', 'amplify-table'));
+      const name = await initCDKProject(projRoot, templatePath, cdkVersion);
+      const outputs = await cdkDeploy(projRoot, '--all');
+
+      const updateTemplatePath = path.resolve(path.join(__dirname, 'backends', 'amplify-table', 'updateIndex'));
+      updateCDKAppWithTemplate(projRoot, updateTemplatePath);
+      await cdkDeploy(projRoot, '--all');
+      const { awsAppsyncApiEndpoint: apiEndpoint, awsAppsyncApiKey: apiKey } = outputs[name];
+      expect(apiEndpoint).toBeDefined();
+      expect(apiKey).toBeDefined();
```
Comment (alharris-at) [inline packages/amplify-graphql-api-construct-tests/src/__tests__/amplify-table.test.ts:38]: Could you add some validation which insert and verify data continuity before/after the second deploy?

File: packages/amplify-graphql-api-construct-tests/src/update-cdk-app-with-template.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import path from 'path';
+import { copySync, moveSync } from 'fs-extra';
+
+export function updateCDKAppWithTemplate(cwd: string, templatePath: string): void {
+  const binDir = path.join(cwd, 'bin');
+  copySync(templatePath, binDir, { overwrite: true });
+  moveSync(path.join(binDir, 'app.ts'), path.join(binDir, `${path.basename(cwd)}.ts`), { overwrite: true });
+}
```
Comment (alharris-at) [inline packages/amplify-graphql-api-construct-tests/src/update-cdk-app-with-template.ts:8]: Could you add this to commands.ts`?

File: packages/amplify-graphql-model-transformer/src/resources/amplify-dynamodb-table/amplify-table-manager-lambda/amplify-table-manager-handler.ts [hunk 4]
```diff
@@ -721,17 +721,19 @@ const usePascalCaseForObjectKeys = (obj: { [key: string]: any }): { [key: string
  * @returns Oject with its values converted to the correct form of boolean or number
```
Comment (palpatim) [inline packages/amplify-graphql-model-transformer/src/resources/amplify-dynamodb-table/amplify-table-manager-lambda/amplify-table-manager-handler.ts:721]:
```suggestion
 * @returns Object with its values converted to the correct form of boolean or number
```

File: packages/amplify-graphql-api-construct-tests/src/__tests__/amplify-table.test.ts (hunk 2 above)
Comment (palpatim) [inline packages/amplify-graphql-api-construct-tests/src/__tests__/amplify-table.test.ts:38]: Should we also assert that the GSI has been created, new table added, new properties set, etc?

File: packages/amplify-graphql-model-transformer/src/resources/amplify-dynamodb-table/amplify-table-manager-lambda/amplify-table-manager-handler.ts [hunk 5]
```diff
@@ -228,7 +228,7 @@ export const onEvent = async (event: AWSCDKAsyncCustomResource.OnEventRequest):
 export const isComplete = async (
   event: AWSCDKAsyncCustomResource.IsCompleteRequest,
 ): Promise<AWSCDKAsyncCustomResource.IsCompleteResponse> => {
-  log('got event', event);
+  log('got event', { ...event, ResponseURL: '...' });
```
Comment (palpatim) [inline packages/amplify-graphql-model-transformer/src/resources/amplify-dynamodb-table/amplify-table-manager-lambda/amplify-table-manager-handler.ts:231]: Here too 
```suggestion
  log('got event', { ...event, ResponseURL: '[redacted]' });
```

Comment (palpatim) [reply packages/amplify-graphql-model-transformer/src/resources/amplify-dynamodb-table/amplify-table-manager-lambda/amplify-table-manager-handler.ts:231]: (Safe to do this in a subsequent PR)

**Review by palpatim — DISMISSED** [review]

**Review by alharris-at — APPROVED** [review]

**Review by palpatim — APPROVED** [review]

**Pull request merged**